cli/crates/backend/src/project.rs

```diff
@@ -96,95 +96,104 @@ pub enum Template<'a> {
+                        }
+                    } else {
+                        return Err(BackendError::MalformedTemplateURL(template.to_owned()));
                     }
                 } else {
-                    return Err(BackendError::MalformedTemplateURL(template.to_owned()));
+                    download_github_template(
+                        &grafbase_path,
+                        GitHubTemplate::Grafbase(GrafbaseGithubTemplate { path: template }),
+                    )
+                    .await?;
                 }
-            } else {
-                download_github_template(
-                    &grafbase_path,
-                    GitHubTemplate::Grafbase(GrafbaseGithubTemplate { path: template }),
-                )
-                .await?;
-            }
-
-            if std::fs::read_dir(&grafbase_path)
-                .expect("We must have a valid directory in this point.")
-                .any(|item| {
-                    item.ok()
-                        .and_then(|dir_entry| dir_entry.path().extension().map(|extension| extension == "ts"))
-                        .unwrap_or_default()
-                })
-            {
-                environment::add_dev_dependency_to_package_json(
-                    &project_path,
-                    GRAFBASE_SDK_PACKAGE_NAME,
-                    GRAFBASE_SDK_PACKAGE_VERSION,
-                )
-                .map_err(BackendError::CommonError)?;
-            }
-        }
-        Template::FromDefault(config_type) => {
-            tokio::fs::create_dir_all(&grafbase_path)
-                .await
-                .map_err(BackendError::CreateGrafbaseDirectory)?;
-
-            let dot_env_path = grafbase_path.join(GRAFBASE_ENV_FILE_NAME);
 
-            let schema_write_result = match config_type {
-                ConfigType::TypeScript => {
-                    let schema_path = grafbase_path.join(GRAFBASE_TS_CONFIG_FILE_NAME);
-
-                    let add_sdk = environment::add_dev_dependency_to_package_json(
+                if std::fs::read_dir(&grafbase_path)
+                    .expect("We must have a valid directory in this point.")
+                    .any(|item| {
+                        item.ok()
+                            .and_then(|dir_entry| dir_entry.path().extension().map(|extension| extension == "ts"))
+                            .unwrap_or_default()
+                    })
+                {
+                    environment::add_dev_dependency_to_package_json(
                         &project_path,
                         GRAFBASE_SDK_PACKAGE_NAME,
                         GRAFBASE_SDK_PACKAGE_VERSION,
                     )
-                    .map_err(Into::into);
+                    .map_err(BackendError::CommonError)?;
+                    return Ok(ConfigType::TypeScript);
+                }
+
+                Ok(ConfigType::GraphQL)
```

Comment:
This diff looks huge, but these return values are the only non-indentation change